Libraries/Opc.Ua.PubSub/Transport/MqttClientCreator.cs

```diff
@@ -101,7 +101,7 @@ internal class MqttClientCreator
 
             while (mqttClient.IsConnected == false)
             {
-                Connect(reconnectInterval, mqttClientOptions, mqttClient);
+                await Connect(reconnectInterval, mqttClientOptions, mqttClient);
```

Comment:
@cristipogacean Yes it can, but the problem here is that the the mqttClient.ConnectAsync called in the Connect method does not set the correct/expected MqttClientConnectResultCode code in case a configuration error is present thus distinguishing between it and a non configuration error such as the server( MQTT broker) not being available becomes impossible. For ex: trying to connect to a broker using an unsupported MQTT protocol version simply throws an exception and does not return as one would expect MqttClientConnectResultCode.ProtocolError. The desired effect of this loop is to allow a connection to connect if the settings are correct and the broker becomes available even after the connection is initiated, of course the undesired effect is that it will loop infinitely if the broker is unreachable or the configuration contains an error.